plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java

```diff
@@ -83,27 +84,24 @@ public List<Route> routes() {
 
     @Override
     public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
-        String algorithm = request.param(PARAMETER_ALGORITHM);
+        String userAlgorithm = request.param(PARAMETER_ALGORITHM);
         String modelId = getParameterId(request, PARAMETER_MODEL_ID);
         Optional<FunctionName> functionName = modelManager.getOptionalModelFunctionName(modelId);
 
-        if (algorithm == null && functionName.isPresent()) {
-            algorithm = functionName.get().name();
-        }
-
-        if (algorithm != null) {
-            MLPredictionTaskRequest mlPredictionTaskRequest = getRequest(modelId, algorithm, request);
+        if (userAlgorithm != null && functionName.isPresent()) {
```

Comment:
the edge case of this implementation is:

when customer didn't provide any algorithm in the request and the model is in cache, it will still go to the next block and make a query to the index, which is unnecessary.